huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go

```diff
@@ -0,0 +1,437 @@
+				"resourceSize":          utils.ValueIngoreEmpty(size),
+				"resourceSizeMeasureId": 30,
+			})
+		}
+	}
+
+	return rst
+}
+
+func buildPayOrderBodyParams(orderId string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"order_id":     orderId,
+		"use_coupon":   "NO",
+		"use_discount": "NO",
+	}
+	return bodyParams
+}
+
+func resourceDscInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getDscInstance: Query the DSC instance
+	var (
+		getDscInstanceHttpUrl = "v1/{project_id}/period/product/specification"
+		getDscInstanceProduct = "dsc"
+	)
+	getDscInstanceClient, err := config.NewServiceClient(getDscInstanceProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DscInstance Client: %s", err)
+	}
+
+	getDscInstancePath := getDscInstanceClient.Endpoint + getDscInstanceHttpUrl
+	getDscInstancePath = strings.ReplaceAll(getDscInstancePath, "{project_id}", getDscInstanceClient.ProjectID)
+
+	getDscInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getDscInstanceResp, err := getDscInstanceClient.Request("GET", getDscInstancePath, &getDscInstanceOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DscInstance")
+	}
+
+	getDscInstanceRespBody, err := utils.FlattenResponse(getDscInstanceResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	dscOrder, err := jmespath.Search("orderInfo[?productInfo.resourceType=='hws.resource.type.dsc.base']",
+		getDscInstanceRespBody)
+	if err != nil {
+		return diag.Errorf("error getting the instance info: base info not found in API response")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("period_unit", parsePeriodUnit(utils.PathSearch("[0].periodType", dscOrder, nil))),
+		d.Set("period", utils.PathSearch("[0].periodNum", dscOrder, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func parsePeriodUnit(periodType interface{}) string {
+	pUnit := fmt.Sprintf("%v", periodType)
+	if pUnit == "3" {
+		return "year"
+
+	} else {
+		return "month"
+	}
+}
+
+func resourceDscInstanceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+
+	if err := common.UnsubscribePrePaidResource(d, config, []string{d.Id()}); err != nil {
+		return diag.Errorf("Error unsubscribing DSC order = %s: %s", d.Id(), err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETE"},
+		Refresh:      waitForBmsInstanceDelete(ctx, d, meta),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        20 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.DiagErrorf("Error deleting DSC instance: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("Error deleting DSC instance: %s", err)
```

please does not use **fmtp** package in new resource

Reply:
done